tools/cfn2ts/lib/codegen.ts

```diff
@@ -339,6 +348,30 @@ export default class CodeGenerator {
     this.code.closeBlock();
   }
 
+  /**
+   * Emit the function that is going to implement the IInspectable interface.
+   *
+   * The generated code looks like this:
+   * public inspect(inspector: cdk.ITreeInspector) {
+   *     inspector.attribute("aws:cdk:cloudformation:type", CfnManagedPolicy.CFN_RESOURCE_TYPE_NAME);
+   *     inspector.attribute("aws:cdk:cloudformation:props", this.cfnProperties);
+   * }
+   *
+   */
+  private emitTreeAttributes(resource: genspec.CodeName): void {
```

Comment:
why not implement this in `CfnResource`?

Reply:
I initially started out trying that, but the main reason for the shift was the properties, each generated class has it's own implementation of `cfnProperties` which is being emitted here. 

Is there another way past that I haven't considered?

Reply:
Ok let’s leave it like this for now